app/models/customer.rb

```diff
@@ -1,4 +1,17 @@
 class Customer < ApplicationRecord
    has_many :invoices
+   has_many :transactions, through: :invoices
    validates_presence_of :first_name, :last_name
+
+   def self.top_five_successful_transactions
+    self.joins(:transactions)
+        .where("transactions.result = 1")
+        .group(:id)
+        .order("COUNT(transactions.id) DESC")
+        .limit(5)
+   end
+   
+   def count_successful_transactions
+    self.transactions.count
```

Comment:
Interesting calling this on a @top_customer iteration customer as a separate method in the view's each block.  AR on AR is possible!